### PR TITLE
Turn ICryptoUtils.createHash into an async function

### PR DESCRIPTION
**Pull request by DonJayamanne**

### CODE REVIEW

File: src/platform/common/types.ts [hunk 1]
```diff
@@ -288,7 +288,7 @@ export interface ICryptoUtils {
      * @param data The string to hash
      * @param [algorithm]
      */
-    createHash(data: string, algorithm?: 'SHA512' | 'SHA256'): string;
+    createHash(data: string, algorithm?: 'SHA512' | 'SHA256'): Promise<string>;
```
Comment (DonJayamanne) [inline src/platform/common/types.ts:291]: This is the change, from sync to async

File: src/platform/common/crypto.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ import * as hashjs from 'hash.js';
  */
 @injectable()
 export class CryptoUtils implements ICryptoUtils {
-    public createHash(data: string, algorithm: 'SHA512' | 'SHA256' = 'SHA256'): string {
+    public async createHash(data: string, algorithm: 'SHA512' | 'SHA256' = 'SHA256'): Promise<string> {
```
Comment (amunger) [inline src/platform/common/crypto.ts:17]: We're just pretending it's async? Why would we want to do this?

Comment (DonJayamanne) [reply src/platform/common/crypto.ts:17]: next pr will add the real bits, wanted to break this work into smaller chunks. one pr already touched almost 70 files

Comment (DonJayamanne) [reply src/platform/common/crypto.ts:17]: i merged another pr into this, this pr now has 74 files changed.
hopefully it makes sense why im trophy to break the work into multiple prs.

again, another pr will add the others changes, adding that here will only make this much larger and more difficult to review

File: src/platform/common/crypto.ts [hunk 3]
```diff
@@ -14,12 +14,12 @@ import * as hashjs from 'hash.js';
...
         return computeHash(data, algorithm);
     }
 }
 
-export function computeHash(data: string, algorithm: 'SHA512' | 'SHA256' | 'SHA1') {
+export async function computeHash(data: string, algorithm: 'SHA512' | 'SHA256' | 'SHA1') {
```
Comment (DonJayamanne) [inline src/platform/common/crypto.ts:22]: This is the other change, changed from sync to async.
This change was supposed to be in another PR, but was accidentally meged into this, now we have 74 file changes here.

**Review by IanMatthewHuff — APPROVED** [review]
It's kinda a pain that we have to do this (wish there was an approved sync option, seems like it should be a small operation). But code is fine.

**Comment by DonJayamanne** [comment]
> wish there was an approved sync option, seems like it should be a small operation

agreed
